frontend/src/metabase/admin/settings/notifications/EditWebhookModal.tsx

```diff
@@ -0,0 +1,88 @@
+import { useMemo } from "react";
+import { t } from "ttag";
+
+import {
+  useDeleteChannelMutation,
+  useEditChannelMutation,
+} from "metabase/api/channel";
+import { Modal } from "metabase/ui";
+import type { NotificationChannel } from "metabase-types/api";
+
+import {
+  handleFieldError,
+  WebhookForm,
+  type WebhookFormProps,
+} from "./WebhookForm";
+
+interface CreateWebhookModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  channel: NotificationChannel;
+}
+
+export const EditWebhookModal = ({
+  isOpen,
+  onClose,
+  channel,
+}: CreateWebhookModalProps) => {
+  const [editChannel] = useEditChannelMutation();
+  const [deleteChannel] = useDeleteChannelMutation();
+
+  const handleSumbit = async (vals: WebhookFormProps) => {
+    return editChannel({
+      id: channel.id,
+      name: vals.name,
+      description: vals.description,
+      details: {
+        url: vals.url,
+        "auth-method": vals["auth-method"],
+        "auth-info": vals["auth-info"],
+      },
+    })
+      .unwrap()
+      .then(() => {
+        onClose();
+      })
+      .catch(e => {
+        handleFieldError(e);
+      });
+  };
+
+  const handleDelete = async () => {
+    await deleteChannel(channel.id).unwrap();
+
+    onClose();
+  };
+
+  const initialValues = useMemo(
+    () => ({
+      url: channel.details.url,
+      name: channel.name,
+      description: channel.description,
+      "auth-method": channel.details["auth-method"],
+      "auth-info": channel.details["auth-info"] || { "": "" },
+    }),
+    [channel],
+  );
+
+  return (
+    <Modal.Root opened={isOpen} onClose={onClose} size="36rem">
+      <Modal.Overlay />
+      <Modal.Content p="1rem">
+        <Modal.Header mb="1.5rem">
+          <Modal.Title>{t`Edit this webhook`}</Modal.Title>
+          <Modal.CloseButton />
+        </Modal.Header>
+        <Modal.Body>
+          <WebhookForm
+            onSubmit={handleSumbit}
+            onCancel={onClose}
+            onDelete={handleDelete}
+            initialValues={initialValues}
+            submitLabel={t`Save changes`}
+          />
+        </Modal.Body>
+      </Modal.Content>
+    </Modal.Root>
```

Comment:
Why do we need to customize the modal like this? can we make plain old `<Modal />` work? I think we want to try to push back on design to use more standard components wherever possible.

Also, we've got some weird scrolling behavior here. Can we put the scroll on the form container?

![scrollweird](https://github.com/user-attachments/assets/6ba29064-0a8c-40c2-af91-318fa6d11672)

Reply:
I'm game to push back, but currently we can't set custom padding on modals because of https://github.com/metabase/metabase/blob/master/frontend/src/metabase/ui/components/overlays/Modal/Modal.styled.tsx#L37. Basically, if you want padding that isn't this, you need to compose the modal manually. Probably something worth untangling

Reply:
Addressed the scrolling otherwise though

Reply:
I'm mad about this. Not your fault, but this is a normal-ass modal, our regular modal component should be fine here. I'm going to go bug designers about it.